### PR TITLE
add ` time.time_ns()`  function

### PR DESCRIPTION
**Pull request by Regexp96**
I add `time.time_ns()` 
` time_ns`  is to get the time in nanoseconds since the epoch

### CODE REVIEW

File: vm/src/stdlib/time_module.rs [hunk 1]
```diff
@@ -58,6 +58,18 @@ pub fn get_time() -> f64 {
     Date::now() / 1000.0
 }
 
+#[cfg(any(not(target_arch = "wasm32"), target_os = "wasi"))]
```
Comment (youknowone) [inline vm/src/stdlib/time_module.rs:61]: i didn't know wasm has confusing discussion about it.
https://stackoverflow.com/questions/6002808/is-there-any-way-to-get-current-time-in-nanoseconds-using-javascript

File: vm/src/stdlib/time_module.rs [hunk 2]
```diff
@@ -58,6 +58,18 @@ pub fn get_time() -> f64 {
...
+pub fn get_time_ns() -> u64 {
```
Comment (youknowone) [inline vm/src/stdlib/time_module.rs:62]: In general, we never want the interpreter is crashed by python code. The correct signature of failible function is like this:
```suggestion
pub fn get_time_ns() -> PyResult<u64> {
```

**Review by youknowone — CHANGES_REQUESTED** [review]

File: vm/src/stdlib/time_module.rs [hunk 3]
```diff
@@ -58,6 +58,18 @@ pub fn get_time() -> f64 {
...
+    match SystemTime::now().duration_since(UNIX_EPOCH) {
+        Ok(v) => v.as_nanos() as u64,
+        Err(err) => panic!("Time error: {:?}", err),
```
Comment (youknowone) [inline vm/src/stdlib/time_module.rs:65]:
```suggestion
        Err(err) => unsafe { std::hint::unreachable_unchecked() }, // guaranted to be not to be happen with now() + UNIX_EPOCH
```

this hint will remove the branch itself for release build

**Review by youknowone — CHANGES_REQUESTED** [review]

File: vm/src/stdlib/time_module.rs [hunk 4]
```diff
@@ -267,5 +279,6 @@ pub fn make_module(vm: &VirtualMachine) -> PyObjectRef {
         "struct_time" => struct_time_type,
         "time" => named_function!(ctx, time, time),
         "perf_counter" => named_function!(ctx, time, time), // TODO: fix
+        "time_ns" => named_function!(ctx,time,time_ns),
```
Comment (youknowone) [inline vm/src/stdlib/time_module.rs:282]: `extend_module!` macro will be helpful to quarantine this function from wasm

**Comment by youknowone** [comment]
@Regexp96 it looks good, but there is a conflict by recent merge. would you rebase it once again?

**Review by youknowone — APPROVED** [review]

**Comment by coolreader18** [comment]
Ah, should've commented earlier, but isn't it possible for `::now() < UNIX_EPOCH` if someone sets the system time to before 1970? I'd think it should just return a negative number in that case (which is definitely possible, you can just do `Err(e) => e.duration()`), though I haven't looked at what cpython does

**Pull request merged**